src/Raven.Server/SqlMigration/RavenDocumentFactory.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Raven.Server.SqlMigration
+{
+    public class RavenDocumentFactory
+    {
+        public WriteOptions Options { get;}
+
+        public RavenDocumentFactory() : this(new WriteOptions()) {}
+
+        public RavenDocumentFactory(WriteOptions options)
+        {
+            Options = options;
+        }
+
+        public RavenDocument FromReader(SqlReader reader, SqlTable table, bool valuesOnly = false)
+        {
+            return FromReader(reader, table, out _, valuesOnly);
+        }
+
+        public RavenDocument FromReader(SqlReader reader, SqlTable table, out Dictionary<string, byte[]> attachments, bool valuesOnly = false)
+        {
+            attachments = new Dictionary<string, byte[]>();
```

Comment:
In most cases, this is creating a dictionary that would be thrown away, empty.
Better to only create this if there is an actual need.